support/sier-codec/src/builder.rs

```diff
@@ -0,0 +1,160 @@
+use crate::{FieldDef, Object, StructDef, Type, Value};
+
+use std::collections::HashMap;
+
+pub struct Builder<'s> {
+    struct_def: &'s StructDef,
+    field_values: HashMap<String, Value>,
+}
+
+impl<'s> Builder<'s> {
+    pub(crate) fn new(struct_def: &'s StructDef) -> Self {
```

Comment:
Why the use of `pub(crate)` here?

Reply:
Because this method is used in the StructDef#builder method. I don't want it to be full public, but it can't be private.